uniflow/op/model/model_server.py

```diff
@@ -474,3 +476,201 @@ def __call__(self, data: List[str]) -> List[str]:
+
+            super().__init__(model_config)
+            self._model_config = BedrockModelConfig(**self._model_config)
+
+            aws_profile = self._model_config.aws_profile
+            aws_region = self._model_config.aws_region if self._model_config.aws_region else None
+            session = boto3.Session(profile_name=aws_profile)
+
+            self._client = session.client("bedrock-runtime", region_name=aws_region)
+
+        except ImportError:
+            raise ModuleNotFoundError(
+                "Failed to import the 'boto3' Python package. "
+                "Please install it by running `pip install boto3`."
+            )
+        except Exception as e:
+            raise ValueError(
+                "Failed to load credentials for authenticating with the AWS client. "
+                "Please ensure that the specified profile name contains valid credentials."
+            ) from e
+
+    def _preprocess(self, data: List[str]) -> List[str]:
+        """Preprocess data.
+
+        Args:
+            data (List[str]): Data to preprocess.
+
+        Returns:
+            List[str]: Preprocessed data.
+        """
+        return data
+
+    def _postprocess(self, data: List[str]) -> List[str]:
+        """Postprocess data.
+
+        Args:
+            data (str): Data to postprocess.
+
+        Returns:
+            List[str]: Postprocessed data.
+        """
+        return data
+    
+    def _get_provider(self) -> str:
+        return self._model_config.model_name.split(".")[0]
+    
+    def enforce_stop_tokens(text: str, stop: List[str]) -> str:
+        """Cut off the text as soon as any stop words occur."""
+        return re.split("|".join(stop), text, maxsplit=1)[0]
+    
+    def prepare_input(
+        self, provider: str, prompt: str, model_kwargs: Dict[str, Any]
+        ) -> Dict[str, Any]:
+        def prepare_anthropic_input(prompt: str, model_kwargs: Dict[str, Any]) -> Dict[str, Any]:
+            input_body = {**model_kwargs, "prompt": f"\n\nHuman: {prompt}\n\nAssistant: "}
+            if "max_tokens_to_sample" not in input_body:
+                input_body["max_tokens_to_sample"] = 256
+            return input_body
+
+        def prepare_ai21_cohere_meta_input(prompt: str, model_kwargs: Dict[str, Any]) -> Dict[str, Any]:
+            return {**model_kwargs, "prompt": prompt}
+
+        def prepare_amazon_input(prompt: str, model_kwargs: Dict[str, Any]) -> Dict[str, Any]:
+            return {"inputText": prompt, "textGenerationConfig": {**model_kwargs}}
+
+        def prepare_default_input(prompt: str, model_kwargs: Dict[str, Any]) -> Dict[str, Any]:
+            return {"inputText": prompt}
+
+        provider_input_preparation = {
+            "anthropic": prepare_anthropic_input,
+            "ai21": prepare_ai21_cohere_meta_input,
+            "cohere": prepare_ai21_cohere_meta_input,
+            "meta": prepare_ai21_cohere_meta_input,
+            "amazon": prepare_amazon_input,
+        }
```

Comment:
qq: any metrics regarding model performance?

Reply:
Unfortunately, I currently do not have solid benchmarks or metrics about the performance of these models.